clusterloader2/pkg/measurement/common/api_availability_measurement.go

```diff
@@ -0,0 +1,185 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"context"
+	"fmt"
+	"k8s.io/perf-tests/clusterloader2/pkg/errors"
+	"k8s.io/perf-tests/clusterloader2/pkg/provider"
+	"sync"
+	"time"
+
+	goerrors "github.com/go-errors/errors"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/klog"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement"
+	measurementutil "k8s.io/perf-tests/clusterloader2/pkg/measurement/util"
+	"k8s.io/perf-tests/clusterloader2/pkg/util"
+)
+
+const (
+	apiAvailabilityName = "APIAvailability"
+)
+
+func init() {
+	if err := measurement.Register(apiAvailabilityName, createAPIAvailabilityMeasurement); err != nil {
+		klog.Fatalf("Cannot register %s: %v", apiAvailabilityName, err)
+	}
+}
+
+func createAPIAvailabilityMeasurement() measurement.Measurement {
+	return &apiAvailabilityMeasurement{}
+}
+
+type apiAvailabilityMeasurement struct {
+	isRunning             bool
+	stopCh                chan struct{}
+	hosts                 []string
+	summaries             []measurement.Summary
+	clusterLevelMetrics   *apiAvailabilityMetrics
+	hostLevelMetrics      map[string]*apiAvailabilityMetrics
+	apiAvailabilityOutput *apiAvailabilityOutput
```

Comment:
let's remove this field - this concenptually isn't part of the measurement. See also my comments below.